docs/proposals/groups.md

```diff
@@ -0,0 +1,91 @@
+# Groups
+
+## Problem
+Project administrators need to be able to define groups so they can efficiently manage policy on their projects, however a project administrator is unlikely to be allowed to manage group membership across all projects.  A single project administrator (or set of project administrators) is likely to own multiple related projects (think single department running multiple projects).  To allow group definitions to be shared across multiple related projects, we'll introduce the concept of Groups (global sets of users), Teams (scoped sets of users), and Organizations (things that reference Users and own Teams).
+
+
+## Use Cases
+ 1.  I want to enable a subset of users to be able to organize sets of Users into Teams, but I only want those changes to affect policy evaluation across some projects.
+ 1.  I want to be able to define some global Groups that will affect policy evaluation across all projects.
+
+
+## Groups
+All groups are global.  That means that if UserA is added to GroupB and ProjectC has a rolebinding that says "GroupB is an editor", UserA now has editor permissions in ProjectC.  Groups will be described using an API object that looks like this:
+```
+// Group contains the set of users that belong to a given group.  The name may not contain ":"
+type Group struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the group
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for Group membership determination.
+
+
+## Teams
+Teams are a lot like groups, but they are always scoped to a particular Organization.  They will be cluster scoped resources, but we'll indicate the Organization scope of a team by prefixing every team name with Organization name.
+```
+// Team contains the set of Users that belong to a given team.  The name of a team is required to 
+// to be in the form of "OrganizationName:TeamName".
+type Team struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the team
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for Team membership determination.  Teams are separate objects from Groups to avoid collisions.  A valid Team.Name can never be used as a Group.Name and vice-versa.  This detail is important later on for authorization parity with kube and backwards compatibility with v3 of OpenShift.
```

Comment:
> A team name will be namescoped by the name of the organization it belongs and a group name cannot contain ":". Are you being explicit here or you are talking specifically about the team part of the team name ("OrganizationName:TeamName")?

`Team.Name="org-name:team"`, so it would be "openshift:committers" as an example.   `Group.Name` cannot contain a ":", so there will never be collisions between them.

Keeping Teams separate from Groups makes sense because teams are managed differently, by a different group of people, with different rules for membership, and different rules for usage.  However, to be able to conform to the existing `UserInfo` interface, we need to be able to unambiguously collapse them into a single field for the trip from the authenticator to the authorizer.